### PR TITLE
Tidy up after recent Focused Launch checkout changes

### PR DESCRIPTION
**Pull request by ciampo**
#### Changes proposed in this Pull Request

Following work from #48452:

* Add more details to some of the Focused Launch logic in the form of inline comments
* Add missing `isInIframe` prop for the Step by Step launch context
* Rename mistyped `window.currentSiteId` to `window._currentSiteId` (and fixed related TypeScript type)

#### Testing instructions

Most changes are just comments.

* Open `apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/attach-launch-sidebar.tsx` and verify that the TypeScript error is gone

Related to #48452

### CODE REVIEW

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/attach-launch-sidebar.tsx [hunk 1]
```diff
@@ -54,6 +55,7 @@ registerPlugin( 'a8c-editor-site-launch', {
 						redirectTo: redirectToWpcomPath,
 						openCheckout,
 						getCurrentLaunchFlowUrl,
+						isInIframe: inIframe(),
```
Comment (ciampo) [inline apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/attach-launch-sidebar.tsx:58]: [Changes in 48452](https://github.com/Automattic/wp-calypso/pull/48452/files#diff-20cce9e66c1059231bcccf929f078c7d8f5877a21859175f7795469214ccfb8dR17) introduced this error (the `isInIframe` prop is required by `LaunchContext` but was not provided here), but somehow neither building the site locally, nor the CI checks flagged it.

We should look into having a full automated TypeScript check — pinging @scinos for some guidance (and in case this is related to #49984)

Comment (razvanpapadopol) [reply apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/attach-launch-sidebar.tsx:58]: We've set `isInIframe` to `false` as default [in the commit where we introduced it](https://github.com/Automattic/wp-calypso/pull/48452/commits/cf11a5a0c42418a3163b68f0c43b95c9d312074e) and the app mounted in `attach-launch-sidebar.tsx` is not using it so it's not really a bug. 

I'm not against adding it to have the context consistent across both apps but maybe we should remove the [default](https://github.com/Automattic/wp-calypso/blob/trunk/packages/launch/src/context.ts#L28-L41) to reduce the maintenance effort?

Good to read on the context subject: https://kentcdodds.com/blog/how-to-use-react-context-effectively

Tl;dr: 
- we can do this to avoid TS errors: 
`const LaunchContext = React.createContext< LaunchContext | undefined >( undefined );`
- in unit tests, unlike stated in React docs, we should wrap the components with the necessary context and not rely on this default value

Comment (ciampo) [reply apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/attach-launch-sidebar.tsx:58]: I see why this wasn't a real bug, but it caused a TypeScript error that was never flagged (the only way I came across it was by opening that file in my IDE).

I agree that one way of fixing this is to make the `isInIframe` prop optional — I will go ahead and apply this change in this PR

Comment (razvanpapadopol) [reply apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/attach-launch-sidebar.tsx:58]: > I agree that one way of fixing this is to make the `isInIframe` prop optional — I will go ahead and apply this change in this PR

It's probably not a good idea to do that since we have it's required in Focused Launch 😬 
Let's continue to keep the context using the same interface for both apps and set a value to this property even if it's not used at the moment in this variation.

File: packages/launch/src/focused-launch/index.tsx [hunk 2]
```diff
@@ -52,7 +52,13 @@ const FocusedLaunch: React.FunctionComponent = () => {
 		}
 	}, [ selectedPlanProductId, planProductIdFromCart, setPlanProductId, hasPaidPlan ] );
 
-	// If there is a purchased plan, remove any selected plan from Launch store
+	// The user may have previously used the launch flow to pick a paid plan,
+	// but they may have then purchased that paid plan before launching.
```
Comment (razvanpapadopol) [inline packages/launch/src/focused-launch/index.tsx]:
```suggestion
	// but they may have then purchased that paid plan or other plan before deciding to continue Launch flow.
```

File: packages/launch/src/focused-launch/summary/index.tsx [hunk 3]
```diff
@@ -574,9 +574,11 @@ const Summary: React.FunctionComponent = () => {
 		}
 	}, [ title, showSiteTitleStep, isSiteTitleStepVisible ] );
 
+	// Launch the site directly if Free plan and subdomain are selected.
+	// Otherwise, show checkout as the next step.
 	const handleLaunch = () => {
-		// Launch the site directly if Free plan and subdomain are selected.
-		// Otherwise, show checkout as the next step.
+		// !isSelectedPlanPaid is used instead of !isSelectedPlanFree because
```
Comment (razvanpapadopol) [inline packages/launch/src/focused-launch/summary/index.tsx]:
```suggestion
		// checking for not having a selected paid plan instead of checking for a selected Free plan because
```

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/attach-launch-sidebar.tsx (hunk 1 above)
Comment (ciampo) [reply apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/attach-launch-sidebar.tsx:58]: Ok then, I guess we can keep the code in this PR as is.

> maybe we should remove the default value of the `LaunchContext`

This could be an option.

The more pressing issue, for me, is to understand how we can get TypeScript errors like this flagged in our automated build / CI processes

File: packages/launch/src/launch/index.tsx [hunk 4]
```diff
@@ -43,8 +43,12 @@ const FocusedLaunchModal: React.FunctionComponent< FocusedLaunchModalProps > = (
 
 	React.useEffect( () => {
 		if ( isLaunchImmediately ) {
-			// if there was a plan in cart before redirect to payment processing,
+			// This happens when the user enters the block editor again after
+			// checkout redirection (i.e. when in Calypso, paying with paypal).
+			// If there was a plan in cart before redirect to payment processing,
 			// remove it now since we don't need to reload the page when dismissing Success view
+			// For more details, see
+			// https://github.com/Automattic/wp-calypso/issues/50122
```
Comment (razvanpapadopol) [inline packages/launch/src/launch/index.tsx]:
```suggestion
			// Calling `unsetPlanProductId()` won't be needed after 
			// https://github.com/Automattic/wp-calypso/issues/50185 is fixed.
```

**Comment by ciampo** [comment]
@razvanpapadopol is this PR good to go, or do you see any other changes required?

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/utils.ts [hunk 5]
```diff
@@ -21,6 +21,17 @@ export const redirectToWpcomPath = ( url: string ): void => {
...
+ * If possible, the checkout is opened as a modal without a page redirect (for Focused Launch).
+ * Otherwise, as a fallback, the user is redirected to the /checkout page or the /home page
+ * (in this case, `siteSlug` and `isEcommerce` params are used to construct the redirect url)
+ *
+ * @param siteSlug The slug (id) of the current site.
+ * @param isEcommerce True if the eCommerce plan is going to be in the checkout.
+ * @param onSuccessCallback Called when the checkout opens as a modal and is completed successfully
+ */
 export const openCheckout = (
 	siteSlug: string = window?.currentSiteId?.toString() || '',
```
Comment (yansern) [inline apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/utils.ts]: Just verified. `window.currentSiteId` doesn't exist. `window._currentSiteId` does. Then we can also fix the typings.

Comment (ciampo) [reply apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/utils.ts]: Cleaned up that variable in https://github.com/Automattic/wp-calypso/pull/50182/commits/4d9ec5fe6dffee874d9096f3a5c13c4021b61b2c

**Comment by ciampo** [comment]
@yansern can you have another look?

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/utils.ts (hunk 5 above)
Comment (yansern) [reply apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/utils.ts]: > @yansern can you have another look?

Looks good! :)
